contributors/design-proposals/node-allocatable.md

```diff
@@ -89,61 +73,204 @@ The flag will be specified as a serialized `ResourceList`, with resources define
-behavior is equivalent to the 1.1 behavior with scheduling based on Capacity.
+Initially we will only support CPU and memory, but will eventually support more resources like [local storage](#phase-3) and io proportional weights to improve node reliability.
 
 #### System-Reserved
 
 In the initial implementation, `SystemReserved` will be functionally equivalent to
 [`KubeReserved`](#kube-reserved), but with a different semantic meaning. While KubeReserved
 designates resources set aside for kubernetes components, SystemReserved designates resources set
 aside for non-kubernetes components (currently this is reported as all the processes lumped
-together in the `/system` raw container).
+together in the `/system` raw container on non-systemd nodes).
+
+## Recommended Cgroups Setup
+
+Following is the recommended cgroup configuration for Kubernetes nodes.
+All OS system daemons are expected to be placed under a top level `SystemReserved` cgroup.
+`Kubelet` and `Container Runtime` are expected to be placed under `KubeReserved` cgroup.
+The reason for recommending placing the `Container Runtime` under `KubeReserved` is as follows:
+
+1. A container runtime on Kubernetes nodes is not expected to be used outside of the Kubelet.
+1. It's resource consumption is tied to the number of pods running on a node.
+
+Note that the hierarchy below recommends having dedicated cgroups for kubelet and the runtime to individally track their usage.
+
+/ (Cgroup Root)
+.
++..systemreserved or system.slice (`SystemReserved` enforced here *optionally* by kubelet)
+.        .    .tasks(sshd,udev,etc)
+.
+.
++..kubereserved or kube.slice (`KubeReserved` enforced here *optionally* by kubelet)
+.	 .
+.	 +..kubelet
+.	 .   .tasks(kubelet)
+.        .
+.	 +..runtime
+.	     .tasks(docker-engine, containerd)
+.	 
+.
++..kubepods or kubepods.slice (Node Allocatable enforced here by Kubelet)
+.	 .
+.	 +..PodGuaranteed
+.	 .	  .
+.	 .	  +..Container1
+.	 .	  .        .tasks(container processes)
+.	 .	  .
+.	 .        +..PodOverhead
+.	 .        .        .tasks(per-pod processes)
+.	 .        ...
+.	 .
+.	 +..Burstable
+.	 .	  .
+.	 .	  +..PodBurstable
+.	 .	  .	    .
+.	 .	  .	    +..Container1
+.	 .	  .	    .         .tasks(container processes)
+.	 .	  .	    +..Container2
+.	 .	  .	    .         .tasks(container processes)
+.	 .	  .	    .
+.      	 .     	  .    	    ...
+.	 .	  .
+.	 .	  ...
+.	 .
+. 	 .
+.	 +..Besteffort
+.	 .	  .
+.	 .	  +..PodBesteffort
+.	 .	  .    	    .
+.	 .	  .	    +..Container1
+.	 .	  .	    .         .tasks(container processes)
+.	 .	  .	    +..Container2
+.	 .	  .	    .         .tasks(container processes)
+.	 .	  .	    .
+.      	 .     	  .    	    ...
+.	 .	  .
+. 	 .	  ...
+
+`systemreserved` & `kubereserved` cgroups are expected to be created by users. If Kubelet is creating cgroups for itself and docker daemon, it will create the `kubereserved` cgroup automatically.
```

Comment:
So... the kubelet is creating a cgroup and putting itself in it?  Then moving the docker daemon or containerd to that cgroup as well?